### PR TITLE
Day5 Ex1 RecyclerView

### PR DESCRIPTION
**Pull request by at-ducle**

### CODE REVIEW

File: app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+
+public class FriendsAdapter extends RecyclerView.Adapter<FriendsAdapter.MyViewHolder> {
+    private ArrayList<UserDay5> mUsers;
+    private Context mContext;
+
+    public FriendsAdapter(ArrayList<UserDay5> mUsers, Context mContext) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java]: please change `mUsers` to `users`
also `mContext` to `context`

File: app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @Override
+    public int getItemCount() {
+        return mUsers == null ? 0 : mUsers.size();
+    }
+
+    class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private TextView mTvName;
+        public Button mBtnDoSth;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java]: Please change `public` to `private`

File: app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+        public MyViewHolder(View itemView) {
+            super(itemView);
+            mTvName = (TextView) itemView.findViewById(R.id.tvName);
+            mBtnDoSth = (Button) itemView.findViewById(R.id.btnDoSth);
+            mBtnDoSth.setOnClickListener(this);
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java]: what `mBtnDoSth` mean? please change this `button` name and `id`

File: app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        @Override
+        public void onClick(View v) {
+            if (v.getId() == R.id.btnDoSth) {
+                Log.i("TAG1", "pressed");
+                if (mUsers.get(getAdapterPosition()).isFriend()) {
+                    mUsers.get(getAdapterPosition()).setFriend(false);
+                    notifyDataSetChanged();
+                } else {
+                    mUsers.get(getAdapterPosition()).setFriend(true);
+                    notifyDataSetChanged();
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java]: Please remove `notifyDataSetChanged()` at line 70 and 73, then move it to line 75

File: app/src/main/java/vn/asiantech/internship/day5/FriendsFragment.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+
+public class FriendsFragment extends Fragment {
+    private RecyclerView mRecyclerViewFriends;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/FriendsFragment.java]: Change variable name `mRecyclerViewFriends` to `mRecyclerViewFriend`

File: app/src/main/java/vn/asiantech/internship/day5/UserDay5.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package vn.asiantech.internship.day5;
+
+public class UserDay5 {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/UserDay5.java]: Please change name to `User` and move it to package `model`

File: app/src/main/res/layout/fragment_friends.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recyclerViewFriends"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/fragment_friends.xml]: change id to `recyclerViewFriend`

File: app/src/main/res/layout/activity_friends.xml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <fragment
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_friends.xml]: Please fix `warning`

File: app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+import android.widget.Button;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+
+public class FriendsAdapter extends RecyclerView.Adapter<FriendsAdapter.MyViewHolder> {
+    private ArrayList<UserDay5> mUsers;
+    private Context mContext;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java]: Please remove this variable, if you don't use it

File: app/src/main/res/layout/item_friend.xml [hunk 10]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_friend.xml]: Please change file name to `item_list_friend`

File: app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+    public FriendsAdapter(ArrayList<UserDay5> mUsers, Context mContext) {
+        this.mUsers = mUsers;
+        this.mContext = mContext;
+    }
+
+    @Override
+    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_friend, null);
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/FriendsAdapter.java]: Please fix `warning`

File: app/src/main/java/vn/asiantech/internship/day5/UserDay5.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    private String name;
+    private String description;
+    private boolean isFriend;
+
+    public UserDay5(String name, boolean isFriend) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day5/UserDay5.java]: Please remove all `method un-use`

File: app/src/main/java/vn/asiantech/internship/activity/FriendsActivity.java [hunk 13]
```diff
@@ -0,0 +1,15 @@
+package vn.asiantech.internship.activity;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+import vn.asiantech.internship.R;
+
+public class FriendsActivity extends AppCompatActivity {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/activity/FriendsActivity.java]: Add java doc for this class

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.model.User;
+
+public class FriendsAdapter extends RecyclerView.Adapter<FriendsAdapter.MyViewHolder> {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: Add java doc for this class

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+        if (user.isFriend()) {
+            holder.mBtnAdd.setText(R.string.button_text_friend);
+            holder.mBtnAdd.setBackgroundColor(Color.RED);
+            holder.mBtnAdd.setCompoundDrawablesWithIntrinsicBounds(R.drawable.checked_32, 0, 0, 0);
+        } else {
+            holder.mBtnAdd.setBackgroundColor(Color.GREEN);
+            holder.mBtnAdd.setText(R.string.button_add);
+            holder.mBtnAdd.setCompoundDrawablesWithIntrinsicBounds(android.R.drawable.ic_menu_add, 0, 0, 0);
+        }
+        holder.mBtnAdd.setWidth(50);
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: why `set width(50)` here?

File: app/src/main/java/vn/asiantech/internship/model/User.java [hunk 16]
```diff
@@ -0,0 +1,23 @@
+package vn.asiantech.internship.model;
+
+public class User {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/model/User.java]: Add java doc for this class

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java (hunk 15 above)
Comment (LeDuc38) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: e xóa dòng này rồi a

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+            holder.mBtnAdd.setCompoundDrawablesWithIntrinsicBounds(android.R.drawable.ic_menu_add, 0, 0, 0);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return mUsers == null ? 0 : mUsers.size();
+    }
+
+    class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: add java doc

File: app/src/main/java/vn/asiantech/internship/fragment/FriendsFragment.java [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.adapter.FriendsAdapter;
+import vn.asiantech.internship.model.User;
+
+public class FriendsFragment extends Fragment {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/fragment/FriendsFragment.java]: add java doc

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java (hunk 15 above)
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: @LeDuc38 
Oke a thấy rồi e, cái này comment củ mà

File: app/src/main/java/vn/asiantech/internship/activity/FriendsActivity.java [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+import vn.asiantech.internship.R;
+
+/**
+ * activity have fragment friend
+ */
+public class FriendsActivity extends AppCompatActivity {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/activity/FriendsActivity.java]: Change class name to `FriendActivity`

File: app/src/main/res/layout/activity_friends.xml [hunk 20]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_friends.xml]: Change file name to `activity_friend`

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.model.User;
+
+/**
+ * Custom adapter to show list friends
+ */
+public class FriendsAdapter extends RecyclerView.Adapter<FriendsAdapter.MyViewHolder> {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: Change file name to `FriendAdapter`

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+    private ArrayList<User> mUsers;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: Use `List<>` instead for `ArrayList<>`

File: app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return mUsers == null ? 0 : mUsers.size();
+    }
+
+    /**
+     * custom viewholder item for list friend
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/adapter/FriendsAdapter.java]: Change `viewholder` to `MyViewHolder`

File: app/src/main/java/vn/asiantech/internship/fragment/FriendsFragment.java [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.adapter.FriendsAdapter;
+import vn.asiantech.internship.model.User;
+
+/**
+ * fragment store list friend
+ */
+public class FriendsFragment extends Fragment {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/fragment/FriendsFragment.java]: Change class name to `FriendFragment`

File: app/src/main/java/vn/asiantech/internship/fragment/FriendsFragment.java [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+    private RecyclerView mRecyclerViewFriend;
+    private ArrayList<User> mUsers;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/fragment/FriendsFragment.java]: Use `List<>` instead for `ArrayList <>`

File: app/src/main/res/layout/fragment_friends.xml [hunk 26]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/fragment_friends.xml]: Change file name to `fragment_friend`

File: app/src/main/res/layout/activity_friend.xml [hunk 27]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <fragment
+        android:id="@+id/fragmentFriends"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_friend.xml]: Change id to `fragmentFriend`

**Pull request merged**